ddd-koans/Value Object/Tiny Types/src/Main.kt

```diff
@@ -0,0 +1,37 @@
+sealed class TimeSignature {
+    companion object {
+        private val timeSignatureRegex = "^(\\d{1,2})/(\\d{1,2})$".toRegex()
+
+        fun create(numerator: Int, denominator: Int): TimeSignature {
+            return runCatching {
+                ValidTimeSignature(Numerator(numerator), Denominator(denominator))
+            }.getOrDefault(InvalidTimeSignature)
+        }
+
+        fun of(timeSignature: String): TimeSignature {
+            val groupValues = timeSignatureRegex.matchEntire(timeSignature)?.groupValues
+            return if (groupValues == null) {
+                InvalidTimeSignature
+            } else {
+                val (numerator, denominator) = (groupValues[1] to groupValues[2])
+                create(numerator.toInt(), denominator.toInt())
+            }
+        }
+    }
+}
+
+object InvalidTimeSignature : TimeSignature()
+
+data class ValidTimeSignature(private val numerator: Numerator, private val denominator: Denominator) : TimeSignature()
```

Comment:
Not sure if it was intended but it looks like we are missing a task/theory which will introduce `data class` and elaborate on it. It should be done before this task. Especially that `tiny types` are also represented as `data classes`

Reply:
There is a task in the project to add a summary and data class task. It could be inserted before that one.